src/components/SelectOption/SelectOption.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import usersFromServer from '../../api/users';
+
+interface Props {
+  name: string;
+  value?: string;
+  label: string;
+  placeholder: string;
+  hasError?: boolean;
+  onSelectedUser: React.Dispatch<React.SetStateAction<number>>;
```

Comment:
[CODE STYLE] - The 'onSelectedUser' prop suggests that a user object will be passed, but it actually expects a user ID (number). Consider renaming it to 'onUserIdSelected' or 'onSelectUserId' for better clarity.